doc/source/whatsnew/v1.1.0.rst

```diff
@@ -976,6 +976,7 @@ Indexing
 - Bug in :meth:`DataFrame.loc` with dictionary of values changes columns with dtype of ``int`` to ``float`` (:issue:`34573`)
 - Bug in :meth:`Series.loc` when used with a :class:`MultiIndex` would raise an IndexingError when accessing a None value (:issue:`34318`)
 - Bug in :meth:`DataFrame.reset_index` and :meth:`Series.reset_index` would not preserve data types on an empty :class:`DataFrame` or :class:`Series` with a :class:`MultiIndex` (:issue:`19602`)
+- Bug in :func:`pandas.core.indexes.api.union_indexes` would lead to :meth:`DataFrame.append` sorting columns even when ``sort=False`` is specified (:issue:`35092`)
```

Comment:
you don't need the first part (as this is a private function); move to reshaping section of Bug Fixes